From 3d7688e15d11f1d26e7531fecb41522004634f43 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1s=20Gonz=C3=A1lez?=
 <tomasagustin.gonzalezorlando@arm.com>
Date: Tue, 14 Mar 2023 11:21:37 +0000
Subject: [PATCH] net: zperf: Reduce the scope of input address variables

Reduce the scope of in4_addr_my and in6_addr_my pointer variables
that are currently global, but they are used only inside
tcp_receiver_thread.

Take the occasion to fix a typo in one error message.

Issue-Id: SCM-6210
Signed-off-by: Luca Fancellu <luca.fancellu@arm.com>
Change-Id: Iacd29da8e9db2bbf450afdcc9ad3f5dfff78ada9
Upstream-Status: Backport [main:d485ef0231ada7caaa685caad253b75793c2c9f1]
---
 subsys/net/lib/zperf/zperf_tcp_receiver.c | 43 ++++++++++-------------
 1 file changed, 19 insertions(+), 24 deletions(-)

diff --git a/subsys/net/lib/zperf/zperf_tcp_receiver.c b/subsys/net/lib/zperf/zperf_tcp_receiver.c
index 148a1eade3c..25f2e0a7b68 100644
--- a/subsys/net/lib/zperf/zperf_tcp_receiver.c
+++ b/subsys/net/lib/zperf/zperf_tcp_receiver.c
@@ -22,9 +22,6 @@ LOG_MODULE_DECLARE(net_zperf, CONFIG_NET_ZPERF_LOG_LEVEL);
 #define NET_LOG_ENABLED 1
 #include "net_private.h"
 
-static struct sockaddr_in6 *in6_addr_my;
-static struct sockaddr_in *in4_addr_my;
-
 #if defined(CONFIG_NET_TC_THREAD_COOPERATIVE)
 #define TCP_RECEIVER_THREAD_PRIORITY K_PRIO_COOP(8)
 #else
@@ -112,9 +109,7 @@ static void tcp_server_session(void)
 	}
 
 	if (IS_ENABLED(CONFIG_NET_IPV4)) {
-		const struct in_addr *in4_addr = NULL;
-
-		in4_addr_my = zperf_get_sin();
+		struct sockaddr_in *in4_addr = zperf_get_sin();
 
 		fds[SOCK_ID_IPV4_LISTEN].fd = zsock_socket(AF_INET, SOCK_STREAM,
 							   IPPROTO_TCP);
@@ -126,7 +121,7 @@ static void tcp_server_session(void)
 		if (MY_IP4ADDR && strlen(MY_IP4ADDR)) {
 			/* Use Setting IP */
 			ret = zperf_get_ipv4_addr(MY_IP4ADDR,
-						  &in4_addr_my->sin_addr);
+						  &in4_addr->sin_addr);
 			if (ret < 0) {
 				NET_WARN("Unable to set IPv4");
 				goto use_existing_ipv4;
@@ -134,26 +129,27 @@ static void tcp_server_session(void)
 		} else {
 		use_existing_ipv4:
 			/* Use existing IP */
-			in4_addr = zperf_get_default_if_in4_addr();
-			if (!in4_addr) {
+			const struct in_addr *addr =
+				zperf_get_default_if_in4_addr();
+			if (!addr) {
 				NET_ERR("Unable to get IPv4 by default");
 				goto error;
 			}
-			memcpy(&in4_addr_my->sin_addr, in4_addr,
+			memcpy(&in4_addr->sin_addr, addr,
 				sizeof(struct in_addr));
 		}
 
-		in4_addr_my->sin_port = htons(tcp_server_port);
+		in4_addr->sin_port = htons(tcp_server_port);
 
 		NET_INFO("Binding to %s",
-			 net_sprint_ipv4_addr(&in4_addr_my->sin_addr));
+			 net_sprint_ipv4_addr(&in4_addr->sin_addr));
 
 		ret = zsock_bind(fds[SOCK_ID_IPV4_LISTEN].fd,
 				 (struct sockaddr *)in4_addr_my,
 				 sizeof(struct sockaddr_in));
 		if (ret < 0) {
 			NET_ERR("Cannot bind IPv4 TCP port %d (%d)",
-				ntohs(in4_addr_my->sin_port), errno);
+				ntohs(in4_addr->sin_port), errno);
 			goto error;
 		}
 
@@ -167,9 +163,7 @@ static void tcp_server_session(void)
 	}
 
 	if (IS_ENABLED(CONFIG_NET_IPV6)) {
-		const struct in6_addr *in6_addr = NULL;
-
-		in6_addr_my = zperf_get_sin6();
+		struct sockaddr_in6 *in6_addr = zperf_get_sin6();
 
 		fds[SOCK_ID_IPV6_LISTEN].fd = zsock_socket(AF_INET6, SOCK_STREAM,
 							   IPPROTO_TCP);
@@ -182,7 +176,7 @@ static void tcp_server_session(void)
 			/* Use Setting IP */
 			ret = zperf_get_ipv6_addr(MY_IP6ADDR,
 						  MY_PREFIX_LEN_STR,
-						  &in6_addr_my->sin6_addr);
+						  &in6_addr->sin6_addr);
 			if (ret < 0) {
 				NET_WARN("Unable to set IPv6");
 				goto use_existing_ipv6;
@@ -190,26 +184,27 @@ static void tcp_server_session(void)
 		} else {
 		use_existing_ipv6:
 			/* Use existing IP */
-			in6_addr = zperf_get_default_if_in6_addr();
-			if (!in6_addr) {
+			const struct in6_addr *addr =
+				zperf_get_default_if_in6_addr();
+			if (!addr) {
 				NET_ERR("Unable to get IPv4 by default");
 				goto error;
 			}
-			memcpy(&in6_addr_my->sin6_addr, in6_addr,
+			memcpy(&in6_addr->sin6_addr, addr,
 				sizeof(struct in6_addr));
 		}
 
-		in6_addr_my->sin6_port = htons(tcp_server_port);
+		in6_addr->sin6_port = htons(tcp_server_port);
 
 		NET_INFO("Binding to %s",
-			 net_sprint_ipv6_addr(&in6_addr_my->sin6_addr));
+			 net_sprint_ipv6_addr(&in6_addr->sin6_addr));
 
 		ret = zsock_bind(fds[SOCK_ID_IPV6_LISTEN].fd,
-				 (struct sockaddr *)in6_addr_my,
+				 (struct sockaddr *)in6_addr,
 				 sizeof(struct sockaddr_in6));
 		if (ret < 0) {
 			NET_ERR("Cannot bind IPv6 TCP port %d (%d)",
-				ntohs(in6_addr_my->sin6_port), errno);
+				ntohs(in6_addr->sin6_port), errno);
 			goto error;
 		}
 
-- 
2.39.1

